src/plugins/discover/public/application/components/sidebar/discover_field.tsx

```diff
@@ -108,70 +111,112 @@ export function DiscoverField({
+        size="s"
+        className="dscSidebarItem__action"
+        onClick={(ev: React.MouseEvent<HTMLButtonElement>) => {
+          ev.preventDefault();
+          ev.stopPropagation();
+          toggleDisplay(field);
+        }}
+        data-test-subj={`fieldToggle-${field.name}`}
+        arial-label={addLabelAria}
+      >
+        {addLabel}
+      </EuiButton>
+    );
+  } else if (field.name !== '_source' && selected) {
+    actionButton = (
+      <EuiButton
+        color="danger"
+        className="dscSidebarItem__action"
+        onClick={(ev: React.MouseEvent<HTMLButtonElement>) => {
+          ev.preventDefault();
+          ev.stopPropagation();
+          toggleDisplay(field);
+        }}
+        data-test-subj={`fieldToggle-${field.name}`}
+        arial-label={removeLabelAria}
+      >
+        {removeLabel}
+      </EuiButton>
+    );
+  }
+
   return (
     <>
-      <div
-        className={`dscSidebarField dscSidebarItem ${showDetails ? 'dscSidebarItem--active' : ''}`}
-        tabIndex={0}
-        onClick={() => onShowDetails(!showDetails, field)}
-        onKeyPress={() => onShowDetails(!showDetails, field)}
-        data-test-subj={`field-${field.name}-showDetails`}
-      >
-        <span className="dscSidebarField__fieldIcon">
-          <FieldIcon
-            type={field.type}
-            label={getFieldTypeName(field.type)}
-            scripted={field.scripted}
-          />
-        </span>
-        <span
-          data-test-subj={`field-${field.name}`}
-          title={field.name}
-          className="dscSidebarField__name"
-        >
-          {useShortDots ? wrapOnDot(shortenDottedString(field.name)) : wrapOnDot(field.displayName)}
-        </span>
-        <span>
-          {field.name !== '_source' && !selected && (
-            <EuiButton
-              fill
+      <EuiPopover
+        display="block"
+        button={
+          <EuiKeyboardAccessible>
+            <FieldButton
               size="s"
-              className="dscSidebarItem__action"
-              onClick={(ev: React.MouseEvent<HTMLButtonElement>) => {
-                ev.preventDefault();
-                ev.stopPropagation();
-                toggleDisplay(field);
+              className="dscSidebarItem"
+              isOpen={showDetails}
+              tabIndex={0}
+              onClick={() => {
+                togglePopover();
               }}
-              data-test-subj={`fieldToggle-${field.name}`}
-              arial-label={addLabelAria}
-            >
-              {addLabel}
-            </EuiButton>
-          )}
-          {field.name !== '_source' && selected && (
-            <EuiButton
-              color="danger"
-              className="dscSidebarItem__action"
-              onClick={(ev: React.MouseEvent<HTMLButtonElement>) => {
-                ev.preventDefault();
-                ev.stopPropagation();
-                toggleDisplay(field);
+              onKeyPress={(event: any) => {
```

Comment:
enter key works now 👍 , but when you use tab key, combined with the return key, I was able to open multiple popovers
<img width="610" alt="Bildschirmfoto 2020-07-30 um 15 24 30" src="https://user-images.githubusercontent.com/463851/88928474-26ad7880-d279-11ea-8728-689955f2a58b.png">

Reply:
checked 7.9 in Lens, same behavior, so is this intended? CC: @mbondyra @wylieconlon @flash1293  

![Bildschirmfoto 2020-07-30 um 15 30 34](https://user-images.githubusercontent.com/463851/88928943-b5ba9080-d279-11ea-9469-dac3dfb9f75b.png)

Reply:
On that note, I'm not sure we need the `onKeyPress` when using `EuiKeyboardAccessible`. I took it out on Lens and toggling the popovers with the keyboard still works.

Reply:
@myasonik do you have any input? ^

Reply:
Just so other folks have visibility into this too, I'm working with @andreadelrio on fixing a few a11y concerns here:
https://github.com/andreadelrio/kibana/pull/2

Reply:
@kertal We've merged that PR ^ and this is the updated behaviour for keyboard navigation in both Discover and Lens

![discover_a11y](https://user-images.githubusercontent.com/4016496/89335085-5fb86500-d64c-11ea-9bfc-88335aa5bcbb.gif)
![lens_keyboard](https://user-images.githubusercontent.com/4016496/89335099-63e48280-d64c-11ea-91a6-63189872dd8e.gif)

And this is the explanation that Michail added for using `ownFocus` on `EuiPopover`

> By setting ownFocus and moving focus to the popover when it opens, keypresses are now up to the popover to figure out what to do with. So pressing Enter/Space on interactive items inside the popover will activate them, but if you're on the popover container, it won't do anything. Pressing esc will put your focus from the popover back onto the button.